### PR TITLE
Only register checkNoUnusedPin task if current project is root + task fix

### PR DESCRIPTION
**Pull request by rubenfiszel**
This task only makes sense as a rootProject's task

### CODE REVIEW

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineVersions.java [hunk 1]
```diff
@@ -70,14 +70,18 @@ public void apply(Project project) {
...
+            }
+        } else {
+            project.getTasks().register("checkNoUnusedPin", NoUnusedPinCheckTask.class, rootVersionsPropsFile);
         }
+
         project.getTasks().register("checkBomConflict", BomConflictCheckTask.class, rootVersionsPropsFile);
-        project.getTasks().register("checkNoUnusedPin", NoUnusedPinCheckTask.class, rootVersionsPropsFile);
         project.getPluginManager().apply(BasePlugin.class);
         project.getTasks().register("checkVersionsProps",
                 task -> task.dependsOn("checkBomConflict", "checkNoUnusedPin"));
```
Comment (robert3005) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineVersions.java]: isn't this wrong now?

Comment (rubenfiszel) [reply gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineVersions.java]: ?

File: gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/NoUnusedPinCheckTask.java [hunk 2]
```diff
@@ -38,7 +38,9 @@ public NoUnusedPinCheckTask(File propsFile) {
 
     @Input
     public final Set<String> getResolvedArtifacts() {
-        return BaselineVersions.getResolvedArtifacts(getProject());
+        return getProject().getAllprojects().stream()
```
Comment (robert3005) [inline gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/NoUnusedPinCheckTask.java:41]: I am pretty sure that this is infinite recursion as getAllProjects returns the project itself.

**Comment by rubenfiszel** [comment]
That's an interesting remark. I didn't think of it. Fortunately, I tested
it and it works. I also think that would be true only if a project
registered as part of getAllProject after resolving every Gradle fields but
that is probably not true as there is no real reason of requiring that from
the Gradle standpoint.

**Comment by robert3005** [comment]
Nah, I confused the flow - I thought we're returning to the same function but we're not. Sorry.

**Review by robert3005 — APPROVED** [review]

**Review by iamdanfox — APPROVED** [review]
LGTM.  I think now that we've got some non-trivial functionality here, I'd like to get into the pattern of always adding some tests that demonstrate the tasks passing and failing, to make sure we don't unintentionally change stuff!  I think we can add a nebula.test rig in a separate PR.

**Pull request merged**